ONSdigital/census-rm-print-file-service
# Check for duplicate UACs

# Motivation and Context
There are circumstances where action rule messages can get ingested twice, causing duplicate rows on the partial print files, which could then meet its expected row count erroneously. We need to log an error in this instance and quarantine the file.

# What has changed
* Check for duplicate UAC's in the partial print file before continuing with it
* Quarantine any files with duplicate UAC's and log an error

# How to test?
Try manually publishing action rule messages with duplicate UACs in the same batch, you should see the resulting print file gets quarantined instead of being encrypted and sent.

# Links
https://trello.com/c/3OuAQeMC/973-investigate-duplicate-rows-in-print-file-service-5
